Description: Also use LDFLAGS from pkg-config that are not -l or -L
 The concrete failure is that the flag -fopenmp is missing for linbox and fflas-ffpack.
 In principle these flags should be respected for all extensions that use
 pkg-config, also in module_list.py.
Author: Tobias Hansen <thansen@debian.org>
Bug: https://trac.sagemath.org/ticket/27205
Bug-Debian: https://bugs.debian.org/919573

--- a/sage/src/sage/env.py
+++ b/sage/src/sage/env.py
@@ -386,22 +386,27 @@
          'FFLASFFPACK_CFLAGS',
          'FFLASFFPACK_INCDIR',
          'FFLASFFPACK_LIBDIR',
+         'FFLASFFPACK_LIBEXTRA',
          'FFLASFFPACK_LIBRARIES',
          'GIVARO_CFLAGS',
          'GIVARO_INCDIR',
          'GIVARO_LIBDIR',
+         'GIVARO_LIBEXTRA',
          'GIVARO_LIBRARIES',
          'GSL_CFLAGS',
          'GSL_INCDIR',
          'GSL_LIBDIR',
+         'GSL_LIBEXTRA',
          'GSL_LIBRARIES',
          'LINBOX_CFLAGS',
          'LINBOX_INCDIR',
          'LINBOX_LIBDIR',
+         'LINBOX_LIBEXTRA',
          'LINBOX_LIBRARIES',
          'SINGULAR_CFLAGS',
          'SINGULAR_INCDIR',
          'SINGULAR_LIBDIR',
+         'SINGULAR_LIBEXTRA',
          'SINGULAR_LIBRARIES']
     """
     import pkgconfig
@@ -416,6 +421,7 @@
         # passed in CFLAGS
         aliases[var + "INCDIR"] = pc['include_dirs']
         aliases[var + "LIBDIR"] = pc['library_dirs']
+        aliases[var + "LIBEXTRA"] = list(filter(lambda s: not s.startswith(('-l','-L')), pkgconfig.libs(lib).split()))
         aliases[var + "LIBRARIES"] = pc['libraries']
 
     # LinBox needs special care because it actually requires C++11 with
--- a/sage/src/sage/libs/linbox/fflas.pxd
+++ b/sage/src/sage/libs/linbox/fflas.pxd
@@ -1,6 +1,7 @@
 # distutils: extra_compile_args = FFLASFFPACK_CFLAGS
 # distutils: libraries = FFLASFFPACK_LIBRARIES
 # distutils: library_dirs = FFLASFFPACK_LIBDIR
+# distutils: extra_link_args = FFLASFFPACK_LIBEXTRA
 # distutils: language = c++ 
 
 from .givaro cimport Modular_double, Modular_float, Dense, Sparse
--- a/sage/src/sage/libs/linbox/linbox_flint_interface.pxd
+++ b/sage/src/sage/libs/linbox/linbox_flint_interface.pxd
@@ -1,5 +1,6 @@
 # distutils: libraries = LINBOX_LIBRARIES
 # distutils: library_dirs = LINBOX_LIBDIR
+# distutils: extra_link_args = LINBOX_LIBEXTRA
 
 from sage.libs.flint.types cimport fmpz_t, fmpz_mat_t, fmpz_poly_t
 
--- a/sage/src/sage/libs/linbox/linbox.pxd
+++ b/sage/src/sage/libs/linbox/linbox.pxd
@@ -1,6 +1,7 @@
 # distutils: extra_compile_args = LINBOX_CFLAGS
 # distutils: libraries = LINBOX_LIBRARIES
 # distutils: library_dirs = LINBOX_LIBDIR
+# distutils: extra_link_args = LINBOX_LIBEXTRA
 # distutils: language = c++
 
 from libc.stdint cimport uint32_t, uint64_t
